core/src/main/scala/caliban/Value.scala

```diff
@@ -201,6 +210,106 @@ private object ValueCirce {
     })
 }
 
+private object ValueJsoniter {
+  import com.github.plokhotnyuk.jsoniter_scala.core._
+  import com.github.plokhotnyuk.jsoniter_scala.macros._
+
+  implicit val streamValueCodec: JsonValueCodec[ResponseValue.StreamValue] =
+    new JsonValueCodec[ResponseValue.StreamValue] {
+      def decodeValue(in: JsonReader, default: ResponseValue.StreamValue): ResponseValue.StreamValue =
+        default // TODO should I create a decoder for stream values?
+      def encodeValue(x: ResponseValue.StreamValue, out: JsonWriter): Unit =
+        out.writeVal(x.toString)
+      def nullValue: ResponseValue.StreamValue =
+        null
+    }
+
+  implicit val booleanValueCodec: JsonValueCodec[Value.BooleanValue] =
+    new JsonValueCodec[Value.BooleanValue] {
+      def decodeValue(in: JsonReader, default: Value.BooleanValue): Value.BooleanValue =
+        Value.BooleanValue(in.readBoolean())
+      def encodeValue(x: Value.BooleanValue, out: JsonWriter): Unit =
+        out.writeVal(x.value)
+      def nullValue: Value.BooleanValue =
+        null
+    }
+
+  implicit val stringValueCodec: JsonValueCodec[Value.StringValue] =
+    new JsonValueCodec[Value.StringValue] {
+      def decodeValue(in: JsonReader, default: Value.StringValue): Value.StringValue =
+        Value.StringValue(Option(in.readString(null)).get)
+      def encodeValue(x: Value.StringValue, out: JsonWriter): Unit =
+        out.writeVal(x.value)
+      def nullValue: Value.StringValue =
+        null
+    }
+
+  implicit val listValueCodec: JsonValueCodec[ResponseValue.ListValue] =
+    new JsonValueCodec[ResponseValue.ListValue] {
+      def decodeValue(in: JsonReader, default: ResponseValue.ListValue): ResponseValue.ListValue =
+        if (in.isNextToken('[')) { // TODO test this
+          val values: mutable.Buffer[ResponseValue] = mutable.ListBuffer[ResponseValue]()
+          if (!in.isNextToken(']')) {
+            in.rollbackToken()
+            var l = -1
+            while (l < 0 || in.isNextToken(',')) {
+              l = 0
+              val responseValue = responseValueCodec.decodeValue(in, responseValueCodec.nullValue)
+              values.append(responseValue)
+            }
+          }
+          ResponseValue.ListValue(values.toList)
+        } else {
+          in.readNullOrTokenError(default, '[')
+        }
+      def encodeValue(x: ResponseValue.ListValue, out: JsonWriter): Unit = {
+        out.writeArrayStart()
+        x.values.foreach(x1 => out.writeRawVal(writeToString(x1).getBytes("UTF-8")))
```

Comment:
Can be simplified to `x.values.foreach(x1 => responseValueCodec.encodeValue(x1, out))`, isn't?